### PR TITLE
Emma PR

### PR DESCRIPTION
**Pull request by emmavanoss**

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    session[:name] = params[:name]
+    @name = session[:name]
+    erb(:welcome)
+  end
+
+  post '/play' do
+    @name = session[:name]
+    @move = params[:move]
+    @computer_move = Randomizer.new.move
+    outcome = Judge.new.compare(person: @move, computer: @computer_move)
```
Comment (topherb99) [inline app.rb]: Maybe use of Judge.run would have been appropriate here for refactor? But it is clear what is happening nonetheless

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+require 'sinatra/base'
```
Comment (topherb99) [inline app.rb]: NO README

File: spec/features/play_game_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+feature 'playing the game' do
+
+  scenario 'player move is displayed' do
+    sign_in
+    click_button('ROCK')
+    expect(page).to have_content('Ginny played ROCK')
+  end
+
+  scenario 'computer move is displayed' do
+    allow_any_instance_of(Randomizer).to receive(:move).and_return('ROCK')
```
Comment (topherb99) [inline spec/features/play_game_spec.rb]: Good use of stubs

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    @name = session[:name]
+    erb(:welcome)
+  end
+
+  post '/play' do
+    @name = session[:name]
+    @move = params[:move]
+    @computer_move = Randomizer.new.move
+    outcome = Judge.new.compare(person: @move, computer: @computer_move)
+    if outcome == :pers
```
Comment (topherb99) [inline app.rb]: Possibly too much logic in the controller? Might benefit from being in the model

File: spec/features/play_game_spec.rb (hunk 3 above)
Comment (topherb99) [reply spec/features/play_game_spec.rb]: Although could have benefitted from having the stub at the top of the feature 'playing the game'

File: app.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  enable :sessions
+
+  get '/' do
+    erb(:index)
+  end
+
+  post '/welcome' do
+    session[:name] = params[:name]
+    @name = session[:name]
+    erb(:welcome)
```
Comment (emmavanoss) [inline app.rb]: post requests should redirect to a get for security

File: app.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    @computer_move = Randomizer.new.move
+    outcome = Judge.new.compare(person: @move, computer: @computer_move)
+    if outcome == :pers
+      @winner = @name
+    elsif outcome == :comp
+      @winner = 'Computer'
+    else
+      @winner = 'Nobody'
+    end
+    erb(:play)
```
Comment (emmavanoss) [inline app.rb]: again, redirect to get request before rendering view

**Pull request closed without merging**